datajunction-ui/src/app/pages/NodePage/AddComplexDimensionLinkPopover.jsx

```diff
@@ -0,0 +1,367 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+import * as React from 'react';
+import DJClientContext from '../../providers/djclient';
+import { ErrorMessage, Field, Form, Formik } from 'formik';
+import { FormikSelect } from '../AddEditNodePage/FormikSelect';
+import AddItemIcon from '../../icons/AddItemIcon';
+import { displayMessageAfterSubmit } from '../../../utils/form';
+import LoadingIcon from '../../icons/LoadingIcon';
+import CodeMirror from '@uiw/react-codemirror';
+import { langs } from '@uiw/codemirror-extensions-langs';
+
+export default function AddComplexDimensionLinkPopover({
+  node,
+  dimensions,
+  existingLink = null,
+  isEditMode = false,
+  onSubmit,
+}) {
+  const djClient = useContext(DJClientContext).DataJunctionAPI;
+  const [popoverAnchor, setPopoverAnchor] = useState(false);
+  const ref = useRef(null);
+
+  useEffect(() => {
+    const handleClickOutside = event => {
+      if (ref.current && !ref.current.contains(event.target)) {
+        setPopoverAnchor(false);
+      }
+    };
+    document.addEventListener('click', handleClickOutside, true);
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  }, [setPopoverAnchor]);
+
+  const joinTypeOptions = [
+    { value: 'left', label: 'LEFT' },
+    { value: 'right', label: 'RIGHT' },
+    { value: 'inner', label: 'INNER' },
+    { value: 'full', label: 'FULL' },
+    { value: 'cross', label: 'CROSS' },
+  ];
+
+  const joinCardinalityOptions = [
```

Comment:
Is there some outcome driven by this selection, or we just store it to the link attributes "for future use" ?